src/app/components/product/productItem/ProductItem.test.tsx

```diff
@@ -0,0 +1,45 @@
+import { render } from 'tests';
+import { within } from '@testing-library/react';
+import { Header } from 'ui/header/Header';
+import { AppLocale } from 'context/locale/appLocale.enum';
+import { ProductItem } from './ProductItem';
+import { PRODUCTS_MOCK } from 'tests/mock/productsMock';
+
+describe('ProductItem component', () => {
+  it('renders name of product', () => {
+    const { getByText } = render(<ProductItem productDetails={PRODUCTS_MOCK[0]} />);
+
+    expect(getByText(PRODUCTS_MOCK[0].name)).toBeInTheDocument();
+  });
+
+  it('renders price correctly in English', () => {
+    const { getByTestId } = render(
+      <>
+        <Header />
+        <ProductItem productDetails={PRODUCTS_MOCK[0]} />
+      </>,
+    );
+
+    const priceElement = getByTestId('price');
+
+    expect(within(priceElement).getByText(/PLN/)).toBeInTheDocument();
+    expect(within(priceElement).getByText(/\./)).toBeInTheDocument();
+    expect(within(priceElement).getByText(/^\d{2}$/)).toBeInTheDocument();
+  });
+
+  it('renders price correctly in Polish', () => {
+    const { getByText, getByTestId } = render(
+      <>
+        <Header />
+        <ProductItem productDetails={PRODUCTS_MOCK[0]} />
```

Comment:
Tutaj i powyżej renderujesz ProductItem razem z Headerem. Wiesz co, staramy się testować jedną rzecz na raz. Wiem że używasz go jako zmianę języka, niemniej dobrze by było to zrobić inaczej.
A mianowicie, rozszerzyć trochę customRenderer, żeby można było dodawać initial data przy jego wejściu, mniej więcej tak:
```
type CustomRenderOptions<Q extends Queries = typeof queries> = RenderOptions<Q> & {
  locale?: AppLocale;
};

...

const customRender = (ui: ReactElement, options?: Omit<CustomRenderOptions, 'wrapper'>) => {
...
```
Wtedy te `options` będą miały to pole `locale`, które można dostarczyć tak:
```
render(<ProductItem ... />, { locale: AppLocale.pl });
```
Taki mały trick ;)